rrrspec-client/lib/rrrspec/client/rspec_runner_factory.rb

```diff
@@ -0,0 +1,24 @@
+require 'rrrspec/client/rspec_runner'
+
+module RRRSpec
+  module Client
+    class RSpecRunnerFactory
+      attr_reader :next_factory
+
+      def initialize(successor)
+        @next_factory = successor
+      end
+
+      def create(file_extension)
+        unless Thread.current[:rspec3_runner]
+          Thread.current[:rspec3_runner] = RSpecRunner.new
```

Comment:
Each slave run as a process. RRRSpec is not a multi-thread application. We do not have more than one slave thread in one process. Thus, there cannot be a race condition for a runner.

In general, a test runner cannot be multi-threaded because the testing subject is not always thread-safe.